spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -259,17 +259,19 @@ public SparkContext createSparkContext() {
 
     for (Object k : intpProperty.keySet()) {
       String key = (String) k;
-      Object value = intpProperty.get(key);
-      logger.debug(String.format("SparkConf: key = [%s], value = [%s]", key, value));
-      conf.set(key, (String) value);
+      String val = toString(intpProperty.get(key));
+      if (!key.startsWith("spark.") || !val.trim().isEmpty()) {
```

Comment:
@felixcheung trim() will make whitespace removed. Tests case is not covered all of cases. I make a PR of @bzz and he will fix this.